### PR TITLE
[REL-13574] Implement SDK detection and installation

### PR DESCRIPTION
**Pull request by dakotasanchez**
## Summary

Implements the `Detector` and `Installer` interfaces that were stubbed in the base branch, wiring up real filesystem-based SDK detection and package-manager-based installation into the `ldcli setup` wizard and hidden subcommands.

- **`FileDetector`** — scans the working directory for project indicators and returns language, framework, package manager, recommended SDK ID, and entry point file
- **`PackageInstaller`** — maps SDK IDs to their install commands and shells out to the appropriate package manager
- **`APIClients`** gains `Detector` and `Installer` fields for test injection, with `FileDetector`/`PackageInstaller` as production defaults
- Fixed `install.go` output: `Package: pkg@version` was printing a trailing `@` when `Version` is empty
- Fixed `cmd/root.go` import ordering (`gofmt`)
- Manual-install SDKs (Java, Android, Swift) now return meaningful package identifiers (`com.launchdarkly:launchdarkly-java-server-sdk` etc.) rather than the raw SDK ID

### Detection (`FileDetector`)

`FileDetector.Detect(dir)` runs four probes in priority order against the project directory:

1. **Node.js / React / Next.js** — reads `package.json` and inspects `dependencies` + `devDependencies`:
   - Has `next` → `node-server` SDK, framework = "Next.js"
   - Has `react` → `react-client-sdk`, framework = "React"
   - Otherwise → `node-server` (plain Node)
   - Package manager inferred from lock files: `pnpm-lock.yaml` → pnpm, `yarn.lock` → yarn, default → npm
2. **Go** — presence of `go.mod` → `go-server-sdk`
3. **Python** — presence of `requirements.txt`, `pyproject.toml`, or `setup.py` → `python-server-sdk`
4. **Java** — presence of `pom.xml` (→ mvn) or `build.gradle`/`build.gradle.kts` (→ gradle) → `java-server-sdk`

If nothing matches, returns an error; the wizard surfaces it and halts.

Entry point detection walks a prioritized candidate list per SDK (e.g. for React: `src/App.tsx` → `src/App.jsx` → `src/index.tsx` → ... → `index.js`) and returns the first path that exists on disk, or the last candidate as a suggested path if none do. The returned path is absolute and passed directly to `Initializer.InjectIntoFile`.

### Installation (`PackageInstaller`)

`PackageInstaller.Install(dir, detectResult)` calls `InstallArgs(sdkID, packageManager)` which maps each SDK to its install command:

| SDK | Command |
|-----|---------|
| `react-client-sdk` | `npm/yarn/pnpm install launchdarkly-react-client-sdk` |
| `node-server` | `npm/yarn/pnpm install @launchdarkly/node-server-sdk` |
| `python-server-sdk` | `pip install launchdarkly-server-sdk` |
| `go-server-sdk` | `go get github.com/launchdarkly/go-server-sdk/v7` |
| `ruby-server-sdk` | `gem install launchdarkly-server-sdk` |
| `dotnet-server-sdk` | `dotnet add package LaunchDarkly.ServerSdk` |
| Java / Android / Swift | no command — `Success: false`, no error |

For SDKs with a command, it shells out via `exec.Command` in the project directory and captures combined output. A non-zero exit wraps the error and output into the returned error. For SDKs without a command, it returns `Success: false` with no error so the wizard proceeds (the user still needs to add the dependency manually, but flag creation and code injection remain valid).

### Wizard Integration

```
selectProject → selectEnvironment → fetchEnvDetails
                                          │
                                          ▼
                                      stepDetect  ←── FileDetector.Detect(cwd)
                                          │              → SDKID, PackageManager, EntryPoint
                                          ▼
                                      stepInstall ←── PackageInstaller.Install(cwd, detectResult)
                                          │              → shells out: npm install / go get / pip install / ...
                                          ▼
                                      stepCreateFlag → stepInit → stepWaitForApp → stepVerify → stepDone
```

`detectResult.SDKID` flows into both the installer (command selection) and the initializer (template selection). `detectResult.EntryPoint` is passed directly to `InjectIntoFile`. All five SDKs the detector can return have init templates, so the `stepWaitForApp` file path display is always valid.

## Test plan

- [x] `go build ./...` passes
- [x] `go test ./internal/setup/...` — 26 tests covering all detection scenarios, package managers, entry point fallback, `firstExistingIn` edge cases, `InstallArgs` for all SDK types, and `PackageInstaller` with mock runner
- [x] `go test ./cmd/setup/...` — detect error/success/JSON, install error/success/JSON/version paths
- [x] `go test ./...` — all 28 packages pass
- [x] `gofmt` clean on all changed files
- [x] `golangci-lint` — no new issues introduced (6 pre-existing `errcheck` findings in ari's files)
- [ ] Manual: `ldcli setup detect --path /path/to/react/project` returns language/framework/SDK info
- [ ] Manual: `ldcli setup detect --path /tmp/empty` returns "could not detect" error
- [ ] Manual: `ldcli setup install --sdk-id node-server` runs `npm install @launchdarkly/node-server-sdk`
- [ ] Manual: `ldcli setup install --sdk-id java-server-sdk` returns `Success: false` with package `com.launchdarkly:launchdarkly-java-server-sdk`

**Requirements**

- [x] I have added test coverage for new or changed functionality
- [x] I have followed the repository's pull request submission guidelines
- [x] I have validated my changes against all supported platform versions

---

> [!NOTE]
> **Medium Risk**
> Adds real project detection and shells out to system package managers during `ldcli setup`, which can affect local environments and introduces platform/command-failure edge cases. Wizard flow changes also alter user interaction and error handling paths.
> 
> **Overview**
> `ldcli setup` now uses real implementations for SDK **detection** and **installation**: `internal/setup.FileDetector` infers language/framework/package manager/entry point from project files, and `internal/setup.PackageInstaller` maps SDK IDs to package-manager commands and executes them (with manual-install SDKs returning `Success=false` plus meaningful package identifiers).
> 
> The setup wizard flow is updated to include an explicit **SDK selection step** (prioritizing detected SDK, falling back when detection fails) and to deep-link the final “toggle your flag” URL. Initialization handling is tightened by correctly mapping `android-client-sdk` and by merging Go imports via AST rewriting to avoid breaking `package` declarations.
> 
> CLI wiring and output are adjusted: `cmd.APIClients` gains injectable `Detector`/`Installer` with production defaults, `setup install` no longer prints a trailing `@` when no version is present, and extensive new tests cover detection/install behavior and wizard transitions.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 32f1834a5ba6492e342442af611757aab5f8611e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/setup/detector.go [hunk 1]
```diff
@@ -25,3 +30,163 @@ var _ Detector = StubDetector{}
...
+			Framework:      "React",
+			PackageManager: pm,
+			SDKID:          "react-client-sdk",
+			EntryPoint: filepath.Join(dir, firstExistingIn(dir, []string{
+				"src/App.tsx", "src/App.jsx", "src/App.js",
+				"src/index.tsx", "src/index.jsx", "src/index.js",
+				"index.js",
+			})),
+		}
+	}
```
Comment (ari-launchdarkly) [inline internal/setup/detector.go:114]: can we flesh these out to account for the:

https://launchdarkly.com/docs/sdk/client-side/javascript
https://launchdarkly.com/docs/sdk/client-side/react/react-native

File: internal/setup/detector.go [hunk 2]
```diff
@@ -25,3 +30,163 @@ var _ Detector = StubDetector{}
...
+}
+
+func detectNodePM(dir string) string {
+	if _, err := os.Stat(filepath.Join(dir, "pnpm-lock.yaml")); err == nil {
+		return "pnpm"
+	}
+	if _, err := os.Stat(filepath.Join(dir, "yarn.lock")); err == nil {
+		return "yarn"
+	}
```
Comment (ari-launchdarkly) [inline internal/setup/detector.go:172]:
```suggestion
	if _, err := os.Stat(filepath.Join(dir, "yarn.lock")); err == nil {
		return "yarn"
	}
   if _, err := os.Stat(filepath.Join(dir, "bun.lock")); err == nil {
		return "bun"
	}
```

File: internal/setup/detector.go (hunk 1 above)
Comment (ari-launchdarkly) [inline internal/setup/detector.go:127]:
```suggestion
	if _, ok := allDeps["react"]; ok {
		return &DetectResult{
			Language:       "JavaScript",
			Framework:      "React",
			PackageManager: pm,
			SDKID:          "react-client-sdk",
			EntryPoint: filepath.Join(dir, firstExistingIn(dir, []string{
				"src/App.tsx", "src/App.jsx", "src/App.js",
				"src/index.tsx", "src/index.jsx", "src/index.js",
				"index.js",
			})),
		}
	}
	if _, ok := allDeps["react-native"]; ok {
		return &DetectResult{
			Language:       "JavaScript",
			Framework:      "React Native",
			PackageManager: pm,
			SDKID:          "react-native-client-sdk",
			EntryPoint: filepath.Join(dir, firstExistingIn(dir, []string{
				"src/App.tsx", "src/App.jsx", "src/App.js",
				"src/index.tsx", "src/index.jsx", "src/index.js",
				"index.js",
			})),
		}
	}
	jsClientFrameworks := map[string]string{
	    "backbone":      "Backbone",
	    "svelte":        "Svelte",
	    "vue":           "Vue",
	    "@angular/core": "Angular",
	    "ember-source":  "Ember",
    }
    for dep, framework := range jsClientFrameworks {
	    if _, ok := allDeps[dep]; ok {
		    return &DetectResult{
			    Language:       "JavaScript",
			    Framework:      framework,
			    PackageManager: pm,
			    SDKID:          "js-client-sdk",
			    EntryPoint: filepath.Join(dir, firstExistingIn(dir, []string{
				    "src/App.tsx", "src/App.jsx", "src/App.js",
				    "src/index.tsx", "src/index.jsx", "src/index.js",
				    "src/main.ts", "src/main.js","index.js",
			    })),
		    }
	    }
    }
	
```

File: internal/setup/detector.go [hunk 3]
```diff
@@ -25,3 +30,163 @@ var _ Detector = StubDetector{}
...
+					"main.py", "app.py", "manage.py", "src/main.py",
+				})),
+			}
+		}
+	}
+	return nil
+}
+
+func detectJava(dir string) *DetectResult {
+	for _, indicator := range []string{"pom.xml", "build.gradle", "build.gradle.kts"} {
```
Comment (ari-launchdarkly) [inline internal/setup/detector.go:208]: I'm not as familiar with Java so forgive me if this isn't relevant, but could the `build.gradle` also apply for the Android SDK?

https://launchdarkly.com/docs/sdk/client-side/android

If so, is there any way for us to knowingly distinguish when the Java app is a Java app versus an android app?

File: internal/setup/detector.go [hunk 4]
```diff
@@ -25,3 +30,163 @@ var _ Detector = StubDetector{}
...
+	}
+	if result := detectGo(dir); result != nil {
+		return result, nil
+	}
+	if result := detectPython(dir); result != nil {
+		return result, nil
+	}
+	if result := detectJava(dir); result != nil {
+		return result, nil
+	}
```
Comment (ari-launchdarkly) [inline internal/setup/detector.go:54]: Can we add a condition for detecting iOS and .NET server since those are also part of the top-10?

File: internal/setup/detector.go [hunk 5]
```diff
@@ -25,3 +30,163 @@ var _ Detector = StubDetector{}
...
+	return nil, errors.New("could not detect project language from directory; try specifying --sdk-id manually")
```
Comment (ari-launchdarkly) [inline internal/setup/detector.go:61]: should we provide a list of the sdks that the user can select from?

File: internal/setup/installer.go [hunk 6]
```diff
@@ -24,3 +29,116 @@ var _ Installer = StubInstaller{}
...
+	}
+}
+
+// nodeInstallCmd returns the install command arguments for a Node.js package manager.
+func nodeInstallCmd(pm, pkg string) []string {
+	switch pm {
+	case "yarn":
+		return []string{"yarn", "add", pkg}
+	case "pnpm":
+		return []string{"pnpm", "add", pkg}
```
Comment (ari-launchdarkly) [inline internal/setup/installer.go:130]:
```suggestion
	case "pnpm":
		return []string{"pnpm", "add", pkg}
    case "bun":
		return []string{"bun", "add", pkg}
```

File: internal/setup/installer.go [hunk 7]
```diff
@@ -24,3 +29,116 @@ var _ Installer = StubInstaller{}
...
+	default:
+		return []string{"npm", "install", pkg}
+	}
+}
+
+// resolveNodePM normalises the package manager name, defaulting to "npm".
+func resolveNodePM(pm string) string {
+	switch pm {
+	case "yarn", "pnpm":
```
Comment (ari-launchdarkly) [inline internal/setup/installer.go]:
```suggestion
	case "yarn", "pnpm", "bun":
```

File: internal/setup/installer.go [hunk 8]
```diff
@@ -24,3 +29,116 @@ var _ Installer = StubInstaller{}
...
+		return []string{"gem", "install", pkg}, pkg
+	case "dotnet-server-sdk":
+		pkg = "LaunchDarkly.ServerSdk"
+		return []string{"dotnet", "add", "package", pkg}, pkg
+	// SDKs requiring manual installation — return a meaningful package identifier
+	// so callers can display what the user needs to add.
+	case "java-server-sdk":
+		return nil, "com.launchdarkly:launchdarkly-java-server-sdk"
+	case "android", "android-client-sdk":
+		return nil, "com.launchdarkly:launchdarkly-android-client-sdk"
```
Comment (ari-launchdarkly) [inline internal/setup/installer.go:116]: for this, the android SDK has a couple different ways of setting it depending on whether the user is using Kotlin or Groovy:

https://launchdarkly.com/docs/sdk/client-side/android#install-the-sdk

Is there a way for us to account for the package here?

File: internal/setup/installer.go [hunk 9]
```diff
@@ -24,3 +29,116 @@ var _ Installer = StubInstaller{}
...
+	case "ruby-server-sdk":
+		pkg = "launchdarkly-server-sdk"
+		return []string{"gem", "install", pkg}, pkg
+	case "dotnet-server-sdk":
+		pkg = "LaunchDarkly.ServerSdk"
+		return []string{"dotnet", "add", "package", pkg}, pkg
+	// SDKs requiring manual installation — return a meaningful package identifier
+	// so callers can display what the user needs to add.
+	case "java-server-sdk":
+		return nil, "com.launchdarkly:launchdarkly-java-server-sdk"
```
Comment (ari-launchdarkly) [inline internal/setup/installer.go:114]: for this, the docs point to us having a way to do this in XML and Gradle:

https://launchdarkly.com/docs/sdk/server-side/java#install-the-sdk

Is that something we'd want to be deterministic about?

**Review by ari-launchdarkly — CHANGES_REQUESTED** [review]
This is really great work. I've got questions around whether we can make this a little more deterministic around some of the edges if we're leaning away from an agent to determine some of the dependencies. If it's not abundantly obvious for us, we can also punt on that

File: internal/setup/detector.go (hunk 5 above)
Comment (dakotasanchez) [reply internal/setup/detector.go:61]: yep let's do that

File: internal/setup/detector.go (hunk 3 above)
Comment (dakotasanchez) [reply internal/setup/detector.go:208]: Yeah we could look for an AndroidManifest.xml

File: internal/setup/installer.go (hunk 8 above)
Comment (dakotasanchez) [reply internal/setup/installer.go:116]: We're accounting for Kotlin vs. Groovy, but we should surface to the user that they need to add the `implementation 'com.launchdarkly:launchdarkly-android-client-sdk:5.+'` to their build file. Unless we want to inject code here as well?

File: internal/setup/installer.go (hunk 9 above)
Comment (dakotasanchez) [reply internal/setup/installer.go:114]: We're deterministic about this detection, but we still are prompting the user to copy/paste a snippet for Gradle/Maven, since there's no shell command to import the LD SDK like with the other languages.

File: internal/setup/detector.go (hunk 1 above)
Comment (dakotasanchez) [reply internal/setup/detector.go:114]: It's pretty tough to detect just plain JS with no framework, so maybe the SDK selector can just take care of that case.

Comment (ari-launchdarkly) [reply internal/setup/detector.go:114]: Maybe. I think we can try and detect some of those based on known active frameworks. I think I put in the suggestion downstream

File: internal/setup/detector.go [hunk 10]
```diff
@@ -25,3 +30,202 @@ var _ Detector = StubDetector{}
...
+			Framework:      "React Native",
+			PackageManager: pm,
+			SDKID:          "react-native-client-sdk",
+			EntryPoint: filepath.Join(dir, firstExistingIn(dir, []string{
+				"src/App.tsx", "src/App.jsx", "src/App.js",
+				"src/index.tsx", "src/index.jsx", "src/index.js",
+				"index.js",
+			})),
+		}
+	}
```
[the remark this answers is not in the document]
Comment (ari-launchdarkly) [reply internal/setup/detector.go:127]: Aw ya, that's a problem

File: internal/setup/detector.go (hunk 1 above)
Comment (dakotasanchez) [reply internal/setup/detector.go:114]: Here's what we're doing now:
1. Has next dep → node-server
2. Has react-native dep → react-native
3. Has react dep → react-client-sdk
4. Has backbone/svelte/vue/angular/ember/preact dep → js-client-sdk
5. Fallback: anything else with a package.json → node-server
